tools/xnvme.c

```diff
@@ -656,7 +656,7 @@ sub_dsm(struct xnvmec *cli)
 	dsm_range->llb = llb;
 	dsm_range->slba = slba;
 
-	err = xnvme_nvm_dsm(&ctx, nsid, dsm_range, nr, ad, idw, idr);
+	err = xnvme_nvm_dsm(&ctx, nsid, dsm_range, nr - 1, ad, idw, idr);
```

Comment:
i would prefer that this be left as is, that is just pass ``nr``.
Then, instead change the value of  ``nr`` in its initialization, and possibly add a comment as well.

Reply:
Up at around line 630.

Reply:
Something like noting ``nr = 0; ///< Zero-based value``

Reply:
If you do this, you need to change line 648 to `dsm_range = xnvme_buf_alloc(dev,sizeof(*dsm_range) * (nr +1));`. Does it still make sense to you @safl ?

Reply:
Actually, why is there an `nr` variable to begin with. As I read it, this function will only send a dsm for one range. So it would be better to just remove the `nr` variable and pass a 0 to the xnvme_nvm_dsm call.

Reply:
I agree. I guess I lean towards your latter approach of removing the ``nr`` and just passing 0.